src/main/java/org/jabref/model/groups/TexGroup.java

```diff
@@ -59,9 +59,15 @@ public AbstractGroup deepCopy() {
 
     @Override
     public boolean equals(Object o) {
-        if (this == o) return true;
-        if (o == null || getClass() != o.getClass()) return false;
-        if (!super.equals(o)) return false;
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        if (!super.equals(o)) {
```

Comment:
shouldnt this be instanceof?

Reply:
I think the check `getClass() != o.getClass()` is slightly better here than `instanceof`. We don't want to consider a different type of groups to be equal to this group.

Reply:
As far as I understand the context I'd say lets leave it as it is with `getClass()'